tests/conftest.py

```
@@ -161,3 +162,58 @@ def sample_files_dir(tmp_path: Path) -> Path:
+    Rules:
+
+    - Calling with no arguments clears all known vars.
+    - A value of ``None`` keeps the var unset (same as not providing it).
+    - An empty string ``""`` sets the var to an empty string.
+    - Passing an unknown var name raises ``KeyError``.
+    """
+
+    def _set(**values: str | None) -> None:
+        unknown = set(values) - _KNOWN_PROVIDER_VARS
+        if unknown:
+            raise KeyError(f"Unknown provider env var(s): {unknown!r}")
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)